src/main/java/picard/sam/CreateSequenceDictionary.java

```diff
@@ -196,6 +223,12 @@ protected int doWork() {
             // read reference sequence one by one and write its metadata
             for (ReferenceSequence refSeq = refSeqFile.nextSequence(); refSeq != null; refSeq = refSeqFile.nextSequence()) {
                 final SAMSequenceRecord samSequenceRecord = makeSequenceRecord(refSeq);
+                //add aliases
+                if (aliasesByContig.containsKey(samSequenceRecord.getSequenceName())) {
```

Comment:
As in my previous comment, you can avoid hashing twice by calling `.get()` and then checking for `null` instead of using `containsKey()`. You only need to use `containsKey()` if your map might contain `null` as a value, and you wish to detect that case.